lib/utils.pm

```diff
@@ -1279,12 +1279,12 @@ sub shorten_url {
     my $ua = Mojo::UserAgent->new;
 
     my $tx = $ua->post('s.qa.suse.de' => form => {url => $url, wishId => $args{wishid}});
-    if (my $res = $tx->success) {
-        return $res->body;
+    if ($tx->res->is_success) {
+        return $tx->res->body;
     }
     else {
-        my $err = $tx->error;
-        die "Shorten url got $err->{code} response: $err->{message}" if $err->{code};
+        my $err = $tx->res->error;
+        die "Shorten url got $err->{code} response: $err->{message}" if $tx->res->code;
```

Comment:
I guess, we also need to print `$tx->res->code` in the `die` call. Not sure where `message` property is contained, but that will be undef now here and in the line below. We could also unite those 2 `die` calls, not sure why do we need both, as it's only about printing or not printing return code, as it might be undef and we will attempt to concat it to the string.

Reply:
yes, the code was strange, we can try to follow the documentation in https://metacpan.org/pod/Mojo::UserAgent#get
I read that some server shutdown before sending the code, but I'm not sure if message would be available then, I added an 'else' at the end more generic.